src/components/Upload/Uploader.tsx

```diff
@@ -214,8 +214,45 @@ export default function Uploader(props: Props) {
             addLogLine(`received file upload request`);
             setWebFiles(props.webFileSelectorFiles);
         } else if (props.dragAndDropFiles?.length > 0) {
-            addLogLine(`received drag and drop upload request`);
-            setWebFiles(props.dragAndDropFiles);
+            if (isElectron()) {
+                const main = async () => {
+                    try {
+                        addLogLine(`uploading dropped files from desktop app`);
+                        // check and parse dropped files which are zip files
+
+                        const electronFiles = [
+                            ...(await Promise.all(
+                                props.dragAndDropFiles.map(async (file) => {
```

Comment:
Just a side note: all of these .map, prormise.all functions will slowly slowly start adding up when it comes to performance.
Here, i would return a for loop to identify zip files, and just zipped them to add via promise.

Reply:
But then also we would need to have an async for loop, which would wait for the zipping to complete no?

The same is happening currently too no? We await if the file is a zip to unzip, otherwise, we just push return the file, in which case it just acts like a sync function.

Reply:
The for-loop version would be a cleaner,